tests/user_extractor_test.php

```diff
@@ -159,11 +159,39 @@ public function test_get_user_case_insensitive() {
         global $CFG;
         $this->resetAfterTest();
 
-        // Arrange data
+        // Arrange data.
         $field = $this->add_user_profile_field('vehicleplate', 'text', true);
-        $expecteduser = $this->getDataGenerator()->create_user();
+        $expecteduser = $this->getDataGenerator()->create_user(['idnumber' => 'NB256']);
         profile_save_data((object)['id' => $expecteduser->id, 'profile_field_vehicleplate' => 'HD4999']);
 
+        // First, test get_user using a core field.
+
+        // Should match with same case.
+        $actualuser = user_extractor::get_user('idnumber', 'NB256', true);
+        $this->assertNotFalse($actualuser);
+        $this->assertSame($expecteduser->id, $actualuser->id);
+
+        // Should match with different case.
+        $actualuser = user_extractor::get_user('idnumber', 'nb256', true);
+        $this->assertNotFalse($actualuser);
+        $this->assertSame($expecteduser->id, $actualuser->id);
+
+        // Should not match with different value (obviously).
+        $actualuser = user_extractor::get_user('idnumber', 'Some other value entirely', true);
+        $this->assertFalse($actualuser);
+
+        if (!in_array($CFG->dbtype, ['mysqli', 'mariadb'])) {
```

Comment:
@dmitriim thoughts on this in the tests? - I'm ok with it and there's no easy API that says "does your db type support case sensitivity"

Reply:
There is $DB->get_dbfamily that might be useful here. Perhaps this affects the mysql family. (instead of needing to list out each one)

https://github.com/moodle/moodle/blob/6f58f72/lib/accesslib.php#L5289